test/variable_tokenizer_test.dart

```diff
@@ -61,6 +86,15 @@ void main() {
     expect(desc.namedVariables?.keys, ['x', 'y']);
   });
 
+  test('indexed can use same variable more than once', () {
+    final indexed = InternalQueryDescription.indexed(
+        'SELECT * FROM foo WHERE a = @ OR bar = @ OR b = @1');
```

Comment:
I find this ambiguous: why is this reusing the index#1, instead of `a = @2 OR bar = @3`?

Reply:
I've matched the behavior from sqlite3 since that's a database engine where I know we can use indexed parameters without having to specify the index every time while still being able to re-use variables. sqlite3 behaves the same: Variables with an explicit index get that index, variables without one get one higher than the current maximum.

I think it might be confusing if the index of a variable could change due to another variable appearing later in the query.

Reply:
Ah I see, it makes sense now, I need to read the documentation more carefully :)